internal/common/allow.go

```diff
@@ -0,0 +1,81 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// the ALLOW_FILE file must conform to the following schema:
+// {
```

Comment:
This format is a great start! My only concern is that whenever we add a new distribution, we would have to change the default allow list. Not much work, but surely a bit annoying. I gave this some thought and I wonder if we can add defaults to distro jsons themselves:

Let's imagine adding `restrictedAccess: true` into https://github.com/osbuild/image-builder/blob/main/distributions/rhel-90/rhel-90.json (well, not in RHEL but in Fedora). If a user wants to build a restricted distribution, we would check their allowlist to see if they explicitly have the approval to build such a distribution.

How does it sound? Happy to diccuss more.

Reply:
Sounds great, everything mentioned above has been implemented!